tensorflow/lite/micro/kernels/expand_dims_test.cc

```diff
@@ -187,25 +184,41 @@ TF_LITE_MICRO_TEST(ExpandDimsNegativeAxisTest1) {
   const float golden_data[] = {0.1, -0.8, -1.2, -0.5, 0.9, 1.3};
   int axis_dims[] = {1, 1};
   const int32_t axis_data[] = {-1};
-  int golden_dims[] = {1, 3, 2, 1};
   int output_dims[] = {4, 1, 3, 2, 1};
   tflite::testing::TestExpandDims<float>(input_dims, input_data, axis_dims,
-                                         axis_data, golden_dims, output_dims,
+                                         axis_data, output_dims, output_dims,
                                          golden_data, output_data);
 }
 
-TF_LITE_MICRO_TEST(ExpandDimsInputOutputDimsMismatchShallFail) {
+TF_LITE_MICRO_TEST(ExpandDimsInputOutputDimsMismatch) {
   float output_data[6];
   int input_dims[] = {3, 1, 3, 2};
   const float input_data[] = {0.1, -0.8, -1.2, -0.5, 0.9, 1.3};
+  const float golden_data[] = {0.1, -0.8, -1.2, -0.5, 0.9, 1.3};
   int axis_dims[] = {1, 1};
   const int32_t axis_data[] = {-1};
   // When input dimension is [1, 3, 2] and the axis is -1, the output dimension
   // should be [1, 3, 2, 1] as in the test case ExpandDimsNegativeAxisTest1.
-  // Shuffle the output dimension to make it incorrect so that the EXPAND_DIMS
-  // op would fail at prepare.
+  // Shuffle the output dimension to make it incorrect to test that output
+  // tensor dimensions are computed correctly.
+  int golden_dims[] = {4, 1, 3, 2, 1};
   int output_dims[] = {4, 1, 3, 1, 2};
 
+  tflite::testing::TestExpandDims<float>(input_dims, input_data, axis_dims,
+                                         axis_data, golden_dims, output_dims,
+                                         golden_data, output_data);
+}
+
+TF_LITE_MICRO_TEST(ExpandDimsAxisOutOfRangeShallFailTest1) {
```

Comment:
Nit: rather than having these tests listed as 1 & 2, can we be more specific with the name about what failure condition it is triggering? Seems like 1 is too high of an axis value, while 2 is too low. Maybe ExpandDimsAxisOutOfRangeHighShallFailTest & ExpandDimsAxisOutOfRangeLowShallFailTest.

Reply:
done.